web-api/terraform/api/api-public.tf

```diff
@@ -156,17 +211,22 @@ resource "aws_route53_record" "api_public_route53_regional_record" {
 
 resource "aws_api_gateway_base_path_mapping" "api_public_mapping" {
   api_id      = aws_api_gateway_rest_api.gateway_for_api_public.id
-  stage_name  = aws_api_gateway_deployment.api_public_deployment.stage_name
+  stage_name  = aws_api_gateway_stage.api_public_stage.stage_name
   domain_name = aws_api_gateway_domain_name.api_public_custom.domain_name
 }
 
 resource "aws_api_gateway_method_settings" "api_public_default" {
   rest_api_id = aws_api_gateway_rest_api.gateway_for_api_public.id
-  stage_name  = aws_api_gateway_deployment.api_public_deployment.stage_name
+  stage_name  = aws_api_gateway_stage.api_public_stage.stage_name
   method_path = "*/*"
 
   settings {
-    throttling_burst_limit = 5000
-    throttling_rate_limit = 10000
+    throttling_burst_limit = 5000 // concurrent request limit
+    throttling_rate_limit = 10000 // per second
   }
 }
+
+resource "aws_wafv2_web_acl_association" "api_public_association" {
+  resource_arn = aws_api_gateway_stage.api_public_stage.arn
+  web_acl_arn  = aws_wafv2_web_acl.apis.arn
+}
```

Comment:
This association requires referencing the API stage ARN, which required us to explicitly create the API stage above.